wolfcrypt/src/pkcs7.c

```diff
@@ -1518,8 +1518,9 @@ static int FlattenAttributes(PKCS7* pkcs7, byte* output, EncodedAttrib* ea,
     for (i = 0; i < eaSz; i++) {
         derArr[i] = (FlatAttrib*) XMALLOC(sizeof(FlatAttrib), pkcs7->heap,
                                           DYNAMIC_TYPE_TMP_BUFFER);
+        derArr[i]->data = NULL;
         if (derArr[i] == NULL) {
-            FreeAttribArray(pkcs7, derArr, eaSz);
+            FreeAttribArray(pkcs7, derArr, i);
```

Comment:
If initialized to all 0's then the FreeAttribArray should be okay free'ing all rows